services/core-api/app/flask_jwt_oidc_local/jwt_manager.py

```diff
@@ -0,0 +1,341 @@
+# Copyright © 2021 thor wolpert
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# Format error response and append status code.
+"""JWTManager handles validating JWTs using a minimal configuration for an OIDC server.
+
+The JWKS store is cached and refreshed on a periodic basis.
+"""
+
+import json
+import ssl  # pylint: disable=unused-import # noqa: F401; for local hacks
+from functools import wraps
+
+from cachelib import SimpleCache
+from flask import _request_ctx_stack, current_app, g, jsonify, request
+from jose import jwt
+from six.moves.urllib.request import urlopen
+
+from .exceptions import AuthError
+
+
+class JwtManager:  # pylint: disable=too-many-instance-attributes
+    """Manages the JWT verification and JWKS key lookup."""
```

Comment:
This is the main class that does jwt validation. 

forked and updated to support multi IDP. 

we could push this back to its own lib and remove it from our code base, but it was easier to develop here this way. 

Initially started with that approach but turned out to be complex with multiple pushes and iterations. 
I have a fork here https://github.com/bcgov/flask-jwt-oidc
and published here https://pypi.org/project/flask-jwt-oidc-mds/

this is a good potential improvement for future.